pinot-plugins/pinot-segment-uploader/pinot-segment-uploader-default/src/main/java/org/apache/pinot/plugin/segmentuploader/SegmentUploaderDefault.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.plugin.segmentuploader;
+
+import com.google.common.base.Preconditions;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.pinot.core.util.IngestionUtils;
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.filesystem.PinotFS;
+import org.apache.pinot.spi.ingestion.batch.BatchConfig;
+import org.apache.pinot.spi.ingestion.batch.BatchConfigProperties;
+import org.apache.pinot.spi.ingestion.batch.spec.Constants;
+import org.apache.pinot.spi.ingestion.segment.uploader.SegmentUploader;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Default implementation of {@link SegmentUploader} with support for all push modes
+ * The configs for push are fetched from batchConfigMaps of tableConfig
+ */
+public class SegmentUploaderDefault implements SegmentUploader {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(SegmentUploaderDefault.class);
+
+  private String _tableNameWithType;
+  private BatchConfig _batchConfig;
+
+  @Override
+  public void init(TableConfig tableConfig)
```

Comment:
similar to the prev comment. imo either (a) pass an auth token here / add it as header/param to the actual upload() calls below or (b) add it as a transient property to table config that's injected by the service

Reply:
@fx19880617 @Jackie-Jiang any thoughts about this suggestion? I remember Jackie felt strongly about not having anything more in the init other than `init(tableConfig)`. Wdyt about changing it to `init(tableConfig, authToken)` ? Or changing upload to `upload(segmentTarURI, authToken)` ?
Right now we're kind of on option b. I would expect the caller to put the auth token into the table config and send it over, but not really update it in ZK.

Reply:
I'd suggest passing a context object that can easily be extended later on without messing with the interface again. This could include additional information such as TLS context or any other connection information we may need to add in the future

Reply:
Added authContext param to upload